fullnode_install.sh

```diff
@@ -255,7 +255,7 @@ echo -e "\nRunning playbook..."
 
 ansible-playbook -i inventory -v site.yml -e "memory_autoset=true"
 
-PRIMARY_IP=$(ifconfig | sed -En 's/127.0.0.1//;s/.*inet (addr:)?(([0-9]*\.){3}[0-9]*).*/\2/p')
+PRIMARY_IP=$(hostname -I)
```

Comment:
This is cleaner, indeed. And doesn't require net-tools.

However, in both cases I realized there's a problem if there is more than one IP configured.
It can be that you have multiple IPs on a single interface, or multiple interfaces.

For example I have a node with 2 IP addresses on a single interface.
```sh
# hostname -I
185.10.xxx.xxx 37.34.xxx.xxx 2a01:7c8:aaaf:34e:0000:xx:xxxx:xxxx
# ifconfig | sed -En 's/127.0.0.1//;s/.*inet (addr:)?(([0-9]*\.){3}[0-9]*).*/\2/p'
185.10.xx.xxx
```
And a server with two interfaces output will be:
```
# hostname -I
10.50.0.16 172.17.0.1
```
If using `hostname -i`, hostname must be resolvable, so will not work in all cases.

If using `hostname -I` would need to pipe to `tr ' ' '\n'|head -1`, e.g:
```sh
# hostname -I|tr ' ' '\n'|head -1
10.50.0.16
```

In any case, these are "edge" cases, as most users will have a single interface. Nonetheless, I think it is important to at least pipe to the `|tr ' ' '\n'` and use `hostname -I`. The information outputted to the users specifies that the IP is being detected in best-effort, so its okay to have the wrong IP, but better than having 2 IP's mixed into the output.

Reply:
I guess `hostname -I|tr ' ' '\n'|head -1` will be the better solution without net-tools then.